src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/srm/algebras/BDD.cs

```diff
@@ -194,90 +193,90 @@ public BDD[] TopSort()
             //order the nodes according to their ordinals
             //into the nonterminals array
             var nonterminals = new List<BDD>[Ordinal + 1];
-            var nodes = new List<BDD>();
-            var stack = new Stack<BDD>();
-            var set = new HashSet<BDD>();
+            var sorted = new List<BDD>();
+            var toVisit = new Stack<BDD>();
+            var visited = new HashSet<BDD>();
 
-            stack.Push(this);
+            toVisit.Push(this);
 
-            while (stack.Count > 0)
+            while (toVisit.Count > 0)
             {
-                BDD node = stack.Pop();
+                BDD node = toVisit.Pop();
+                // true and False are not included in the result
                 if (node.IsFull || node.IsEmpty)
                     continue;
 
                 if (node.IsLeaf)
                 {
-                    nodes.Add(node);
+                    // MTBDD terminals can be directly added to the sorted nodes
```

Comment:
Can you add a parenthetical saying why?

Reply:
The reason is that adding leaves here preserves the topological order of nodes. 
The topological order being that a child precedes its parent in the nodes list.
I will not edit it myself but Olli could add that comment.

Reply:
Done!